### PR TITLE
use ark

### PR DESCRIPTION
**Pull request by mburns**
related to https://github.com/racker/chef/pull/3914

### CODE REVIEW

**Comment by shawnashlee** [comment]
reads okay

File: test/unit/spec/server_spec.rb [hunk 1]
```diff
@@ -14,10 +14,6 @@
     expect(chef_run).to install_package('numactl')
   end
 
-  # it 'unpacks cassandra tarball' do
-  #   expect(chef_run).to create_untar_archive('cassandra')
-  # end
-
   it 'downloads cassandra tarball' do
     expect(chef_run).to create_remote_file('/usr/src/cassandra-20160921235957.tar.gz')
```
Comment (drenalin23) [inline test/unit/spec/server_spec.rb]: Isn't this test going to fail now that you have removed the remote_file resource?

File: test/unit/spec/default_spec.rb [hunk 2]
```diff
@@ -14,9 +14,6 @@
     expect(chef_run).to run_execute('install_cqlsh')
   end
 
-  # untar_archive needs to define a custom matcher to be testable here
-  # e.g. https://github.com/sethvargo/chefspec#testing-lwrps
-
   it 'downloads cql tarball' do
     expect(chef_run).to create_remote_file_if_missing('/usr/src/cql-1.0.5.tar.gz')
```
Comment (drenalin23) [inline test/unit/spec/default_spec.rb]: Same I think this will fail with your removals above.

**Comment by drenalin23** [comment]
For those tests you can add a matchers.rb file in spec/support with:

# matcher for install_ark
def install_ark(resource_name)
  ChefSpec::Matchers::ResourceMatcher.new(:ark, :install, resource_name)
end

def put_ark(resource_name)
  ChefSpec::Matchers::ResourceMatcher.new(:ark, :put, resource_name)
end


then you should be able to call those in a test.

**Pull request merged**